exercises/simple-linked-list/simple-linked-list.spec.js

```diff
@@ -0,0 +1,142 @@
+import { List, Element } from './simple-linked-list';
+
+describe('simple-linked-list', () => {
+  test('exports a List constructor', () => {
+    expect(List).toBeDefined();
+  });
+
+  xtest('exports a Element constructor', () => {
+    expect(Element).toBeDefined();
+  });
+
+  describe('Element', () => {
+    xtest('is a constructor', () => {
+      const el = new Element(1);
+      expect(el).toBeDefined();
+
+      expect(() => {
+        Element(1);
+      }).toThrow();
+    });
+
+    xtest('requires an argument', () => {
+      const el = new Element(1);
+      expect(el).toBeDefined();
+
+      expect(() => new Element()).toThrow();
+    });
+
+    xtest('has as a value', () => {
+      const el = new Element(1);
+      expect(el.value).toBe(1);
+    });
+
+    xtest('reference to next is undefined by default', () => {
+      const el = new Element(1);
+      expect(el.next).toBeUndefined();
+    });
+  });
+
+  xtest('accepts a reference to the next element', () => {
+    const elOne = new Element(1);
+    const elTwo = new Element(2, elOne);
+    expect(elTwo.next).toBe(elOne);
+  });
+
+  xtest('requires an instance of Element as next element', () => {
+    expect(() => new Element(1, true)).toThrow();
+    expect(() => new Element(1, 'lorem ipsum')).toThrow();
+    expect(() => new Element(1, [])).toThrow();
+    expect(() => new Element(1, {})).toThrow();
+  });
+
+  describe('List', () => {
+    xtest('is a constructor', () => {
+      const ll = new List();
+      expect(ll).toBeDefined();
+    });
+
+    xtest('allows you to append an element', () => {
+      const ll = new List();
+      const el = new Element(1);
+      ll.push(el);
+      expect(ll.head).toBe(el);
+
+      const ll2 = new List();
+      const elOne = new Element(1);
+      const elTwo = new Element(2);
+      ll2.push(elOne);
+      ll2.push(elTwo);
+      expect(ll2.head.next).toBe(elTwo);
+    });
+
+    xtest('allows you to prepend an element', () => {
+      const ll = new List();
+      const el = new Element(1);
+      ll.unshift(el);
+      expect(ll.head).toBe(el);
+
+      const ll2 = new List();
+      const elOne = new Element(1);
+      const elTwo = new Element(2);
+      ll2.unshift(elOne);
+      ll2.unshift(elTwo);
+      expect(ll2.head).toBe(elTwo);
```

Comment:
append - `push`
prepend - `unshift`
these names are confusing IMO. I would like to head what others think on this. The public API should be intuitive especially when there are these many functions. Since there is no canonical data for this exercise, we will have to take a call.

Reply:
It uses the same API naming as JS arrays, that seems sensible to me. Anyone doing this exercise will know enough JS/ES to know shift/pop & unshift/push, IMO.

Reply:
I'm in favor of using the JS names.  It seems like the prevailing paradigm.